test/unittests/test_analyse_xml_data/test_extract_links_from_XML.py

```diff
@@ -0,0 +1,142 @@
+# pylint: disable=missing-class-docstring,missing-function-docstring,protected-access
+
+import pytest
+from lxml import etree
+from pytest_unordered import unordered
+
+from dsp_tools.analyse_xml_data.extract_links_from_XML import (
+    _create_classes_from_single_resource,
+    _extract_ids_from_one_resptr_prop,
+    _extract_ids_from_one_text_value,
+    _extract_ids_from_text_prop,
+    _get_all_links_from_one_resource,
+)
+
+
+def test_create_classes_from_single_resource() -> None:
+    test_ele = etree.fromstring(
+        '<resource xmlns="https://dasch.swiss/schema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" '
+        'label="res_A_19" restype=":TestThing" id="res_A_19" permissions="res-default"><resptr-prop '
+        'name=":hasResource1"><resptr permissions="prop-default">res_B_19</resptr><resptr '
+        'permissions="prop-default">res_C_19</resptr></resptr-prop><text-prop name=":hasRichtext"><text '
+        'permissions="prop-default" encoding="xml"><a class="salsah-link" href="IRI:res_B_19:IRI">res_B_19</a><a '
+        'class="salsah-link" href="IRI:res_C_19:IRI">res_C_19</a></text></text-prop></resource>'
+    )
+    res_resptr_links, res_xml_links, res_all_used_ids = _create_classes_from_single_resource(test_ele)
+    #########
+    res_B_19 = [obj.object_id for obj in res_resptr_links]  # type: ignore[union-attr]
```

Comment:
You could remove all 3 ignores of this function, if `_create_classes_from_single_resource()` doesn't return `tuple[None, None, None]` (as I suggested above).

Reply:
Plus, you can then make this test function more concise: `assert "res_B_19" in [obj.object_id for obj in res_resptr_links]`

Like this, you have less lines, and you can remove the ugly `#####` lines